src/scaladoc/scala/tools/nsc/doc/html/SyntaxHigh.scala

```diff
@@ -211,6 +211,30 @@ private[html] object SyntaxHigh {
       out.toString
     }
 
+    def escape(str: String): String = {
+      val array = str.toCharArray
+      val len = array.length
+      val buf = new java.lang.StringBuilder(len)
+      @tailrec def loop(i: Int): String = {
+        if (i < len) {
+          (array(i): @switch) match {
```

Comment:
is forcing a switch a good idea here? This would become a lookup switch, and with just 4 cases, I'm not sure whether that's faster than just if/else if/else if/else

Reply:
`@switch` doesn't change the compiler's behavior, it just ensures that a switch is emitted. This is similar to `@tailrec`.